tasks/release/ReleaseVersionController.php

```diff
@@ -36,15 +36,50 @@ function assignVersionToCompletedTickets($version = null) {
     return [$version, join("\n", $output)];
   }
 
-  private function getDoneIssuesWithoutVersion() {
-    $jql = "project = $this->project AND status = Done AND fixVersion = EMPTY AND updated >= -52w";
-    $result = $this->jira->search($jql, ['key']);
-    return array_map(function ($issue) {
-      return [
-        'id' => $issue['id'],
-        'key' => $issue['key'],
-      ];
-    }, $result['issues']);
+  function determineNextVersion() {
+    $last_version = $this->jira->getLastReleasedVersion();
+    $parsed_last_version = VersionHelper::parseVersion($last_version['name']);
+
+    $version_to_increment = VersionHelper::PATCH;
+
+    if ($this->project === JiraController::PROJECT_MAILPOET) {
+      $free_increment = $this->checkProjectVersionIncrement(JiraController::PROJECT_MAILPOET);
+      $premium_increment = $this->checkProjectVersionIncrement(JiraController::PROJECT_PREMIUM);
+
+      if (in_array(VersionHelper::MINOR, [$free_increment, $premium_increment])) {
+        $version_to_increment = VersionHelper::MINOR;
+      }
+    }
+
+    $parsed_last_version[$version_to_increment]++;
```

Comment:
If we increase minor version, then patch should be reset to `0`. I think that's missing here. Maybe a method in the `VersionHelper` could do it?

Reply:
Good catch, I've added a helper method.